src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobMover.java

```diff
@@ -45,27 +41,27 @@ public void moveToRejectedContainer(String blobName, String containerName) {
         if (!sourceBlob.exists()) {
             logger.error("File already deleted. " + loggingContext);
         } else {
-            copy(targetBlob, sourceBlob, loggingContext);
+            String sasToken = sourceBlob
+                .generateSas(
+                    new BlobServiceSasSignatureValues(
+                        OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC),
+                        new BlobContainerSasPermission().setReadPermission(true)
+                    )
+                );
+
+            if (targetBlob.exists()) {
+                targetBlob.createSnapshot();
+            }
+
+            targetBlob
+                .getBlockBlobClient()
+                .copyFromUrl(sourceBlob.getBlockBlobClient().getBlobUrl() + "?" + sasToken);
+
             sourceBlob.delete();
             logger.info("File successfully moved to rejected container. " + loggingContext);
         }
     }
 
-    private void copy(BlobClient targetBlob, BlobClient sourceBlob, String loggingContext) {
-        String sasToken = sourceBlob
-            .generateSas(
-                new BlobServiceSasSignatureValues(
-                    OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC),
-                    new BlobContainerSasPermission().setReadPermission(true)
-                )
-            );
-        targetBlob
-            .getBlockBlobClient()
-            .copyFromUrl(sourceBlob.getBlockBlobClient().getBlobUrl() + "?" + sasToken);
-
-        logger.info("File successfully uploaded to rejected container. " + loggingContext);
```

Comment:
this has gone from right hand side diff. intentional?

Reply:
yeah, this and then "File successfully moved to rejected container." just next to it looked weird.